File: webflux-dialogue/src/testFixtures/java/com/study/webflux/rag/fixture/CreditTransactionFixture.java
package com.study.webflux.rag.fixture;

import com.study.webflux.rag.domain.credit.model.CreditTransaction;
import com.study.webflux.rag.domain.credit.model.CreditTransactionType;
import com.study.webflux.rag.domain.credit.model.source.ConversationDeduction;
import com.study.webflux.rag.domain.credit.model.source.SignupBonus;
import com.study.webflux.rag.domain.dialogue.model.UserId;

public final class CreditTransactionFixture {

	public static final long DEFAULT_AMOUNT = 100L;

	private CreditTransactionFixture() {
	}

	public static CreditTransaction deduction(UserId userId, long balanceBefore) {
		return CreditTransaction.of(
			userId,
			CreditTransactionType.DEDUCT,
			new ConversationDeduction(ConversationSessionFixture.createId()),
			DEFAULT_AMOUNT,
			balanceBefore,
			balanceBefore - DEFAULT_AMOUNT);
	}

	public static CreditTransaction signupBonus(UserId userId, long bonusAmount) {
		return CreditTransaction.of(
			userId,
			CreditTransactionType.CHARGE,
			new SignupBonus(),
			bonusAmount,
			0L,
			bonusAmount);
	}
}

File: backend/domain/src/testFixtures/kotlin/com/miyou/app/fixture/CreditTransactionFixture.kt
package com.miyou.app.fixture

import com.miyou.app.domain.credit.model.ConversationDeduction
import com.miyou.app.domain.credit.model.CreditTransaction
import com.miyou.app.domain.credit.model.CreditTransactionType
import com.miyou.app.domain.credit.model.SignupBonus

object CreditTransactionFixture {
    const val DEFAULT_AMOUNT = 100L

    @JvmStatic
    fun deduction(
        userId: String,
        balanceBefore: Long,
    ): CreditTransaction =
        CreditTransaction.of(
            userId,
            CreditTransactionType.DEDUCT,
            ConversationDeduction(ConversationSessionFixture.createId().value),
            DEFAULT_AMOUNT,
            balanceBefore,
            balanceBefore - DEFAULT_AMOUNT,
        )

    @JvmStatic
    fun signupBonus(
        userId: String,
        bonusAmount: Long,
    ): CreditTransaction =
        CreditTransaction.of(
            userId,
            CreditTransactionType.CHARGE,
            SignupBonus,
            bonusAmount,
            0L,
            bonusAmount,
        )
}
